migrations/versions/bc0ac747cec6_add_run_reason_column.py

```diff
@@ -0,0 +1,32 @@
+"""add run reason column
+
+Revision ID: bc0ac747cec6
+Revises: 5d516a1f293d
+Create Date: 2022-06-07 15:21:28.698451
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "bc0ac747cec6"
+down_revision = "5d516a1f293d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column("run", sa.Column("reason", sa.String(length=250), nullable=True))
+
+    # use some custom logic to backfill the column
+    op.execute("UPDATE run SET reason = 'commit' WHERE name LIKE 'commit%'")
+    op.execute("UPDATE run SET reason = 'pull request' WHERE name LIKE 'pull request%'")
+    op.execute(
+        "UPDATE run SET reason = 'manual commit' WHERE name LIKE 'manual commit%'"
+    )
+    op.execute("UPDATE run SET reason = 'merge commit' WHERE name LIKE '%merge-commit'")
+    op.execute("UPDATE run SET reason = 'nightly' WHERE name LIKE '%nightly'")
```

Comment:
Looking for feedback on whether this backfill code is necessary. I think it would be nice

Reply:
I think it would be nice too. I'm sure @ElenaHenderson will have opinions on it! But I'm glad you included it

Reply:
Yep, we should do this.

Reply:
I decided to wait on the backfill code for now. When we merge and deploy this PR, it will add an empty Reason column. Then we can change all pipelines to start publishing reasons. And finally I'll come back and add a migration that backfills any rows with empty reasons. Thanks for the idea @ElenaHenderson .